piecedirectory/piecedirectory.go

```diff
@@ -432,15 +442,18 @@ func parsePieceWithDataSegmentIndex(pieceCid cid.Cid, unpaddedSize int64, r type
 	panicked := false
 	indexData, err := parseDataSegmentIndex(pieceCid, bufio.NewReaderSize(cr, PodsiBuffesrSize), &panicked)
 	if err != nil {
+		log.Debugw("Failed to parse data segment index", "error", err)
```

Comment:
This additional logging isn't required because the error will be logged here already: https://github.com/filecoin-project/boost/blob/main/piecedirectory/piecedirectory.go#L321